src/test/java/io/confluent/connect/elasticsearch/bulk/BulkProcessorTest.java

```diff
@@ -305,4 +319,84 @@ public void unretriableErrors() throws InterruptedException {
     }
   }
 
+  @Test
+  public void failOnMalformedDoc() throws InterruptedException {
+    final int maxBufferedRecords = 100;
+    final int maxInFlightBatches = 5;
+    final int batchSize = 2;
+    final int lingerMs = 5;
+    final int maxRetries = 3;
+    final int retryBackoffMs = 1;
+    final BehaviorOnMalformedDoc behaviorOnMalformedDoc = BehaviorOnMalformedDoc.FAIL;
+
+    final String errorInfo = " [{\"type\":\"mapper_parsing_exception\",\"reason\":\"failed to parse\"," +
+        "\"caused_by\":{\"type\":\"illegal_argument_exception\",\"reason\":\"object\n" +
+        " field starting or ending with a [.] makes object resolution ambiguous: [avjpz{{.}}wjzse{{..}}gal9d]\"}}]";
+    client.expect(Arrays.asList(42, 43), BulkResponse.failure(false, errorInfo));
+
+    final BulkProcessor<Integer, ?> bulkProcessor = new BulkProcessor<>(
+        new SystemTime(),
```

Comment:
optional: I know it's same as everywhere in this test, but should we use `Time.SYSTEM` instead? Feel free to change everywhere, only in new tests, or not at all. Kind of like [ignore, warn, fail]-style set of options here too :)